release/models/platform/openconfig-platform-pipeline-counters.yang

```diff
@@ -482,6 +483,15 @@ module openconfig-platform-pipeline-counters {
         "The integrated-circuit lookup subsystem block utilization percentage.";
     }
 
+    leaf lookup-utilization-alarm {
+      type boolean;
+      description
+        "The integrated-circuit lookup subsystem has exceeded the threshold that
+        is set for it by the system. In the case of configurable threshold values
+        the threshold value is set within the /system/thresholds hierarchy with
+        the IC_LOOKUP_MEMORY threshold type.";
+    }
+
```

Comment:
I'm kind of reluctant to add more utilization related attributes in the pipeline counters model. I usually think of "counters" as things that are updated by the forwarding plane from packets going through, whereas utilization is more of a capacity planning/monitoring kind of thing and is updated by the management plane; so wondering if there is some more appropriate model to hold that (if it doesn't exist, maybe we could create one, as it seems like a common and useful thing to monitor).

From the producer's point of view, it would also make things easier because the component responsible for gathering counters is typically different from the one(s) managing lookup tables.

The comment is not specific to this particular attribute, I think it's generally true for any kind of hardware lookup mechanism (nexhop, acl, etc).

Reply:
I think we have a fairly common pattern through openconfig that we have things like counters alongside other attributes (consider something like oper-status of an interface vs. interface counters) -- I don't think we'd want to create a whole other subtree for utilisation.

We could, however, move the counters into a `counters` container so it was clear what was utilisation and what wasn't in this model and allow those subtrees to be collected by different infrastructure.

Happy to chat about what causes issues with them being in the same model (I guess I'm curious if same model implies same producer, and if so, with what granularity).

Reply:
It does seem a bit odd to couple `/system/thresholds` with `/components/component/integrated-circuits/pipeline-counters/...`. This would mean that for every `type` of `threshold` change, there would need a corresponding change to the relevant `component` subtree.

I'm opting for creating a utilization subtree within `/system/thresholds/threshold/state/`, this could also contain the `threshold-utilization-alarm` boolean and easily extend to other types of components.